src/main/java/uk/gov/pay/directdebit/mandate/model/GoCardlessMandate.java

```diff
@@ -1,20 +1,21 @@
 package uk.gov.pay.directdebit.mandate.model;
 
+import uk.gov.pay.directdebit.common.model.subtype.gocardless.GoCardlessCreditorId;
+
 public class GoCardlessMandate {
 
     private Long id;
     private final Long mandateId;
     private final String goCardlessMandateId;
     private final String goCardlessReference;
+    private final GoCardlessCreditorId goCardlessCreditorId;
 
-    public GoCardlessMandate(Long id, Long mandateId, String goCardlessMandateId, String goCardlessReference) {
+    public GoCardlessMandate(Long id, Long mandateId, String goCardlessMandateId, String goCardlessReference, GoCardlessCreditorId goCardlessCreditorId) {
         this.id = id;
         this.mandateId = mandateId;
         this.goCardlessMandateId = goCardlessMandateId;
         this.goCardlessReference = goCardlessReference;
-    }
-    public GoCardlessMandate(Long mandateId, String goCardlessMandateId, String goCardlessReference) {
```

Comment:
Ah, I see, you did remove this constructor.

I actually quite like constructors like this because it makes it clear that the `id` can legitimately be `null`.

Reply:
As discussed if we have more than two constructors we should use Builder pattern and we are going to rollback this constructor.